### ISSUE TITLE
Avoid additional heap allocation for fields

### ISSUE THREAD

**Issue by dfordivam** [LedgerHQ/ledger-device-rust-sdk#169]
It may be possible to avoid additional allocation of the fields on heap by making use of APIs like [core](https://doc.rust-lang.org/stable/core/index.html)::[ffi](https://doc.rust-lang.org/stable/core/ffi/index.html)::[CStr](https://doc.rust-lang.org/stable/core/ffi/struct.CStr.html#) [from_bytes_until_nul](https://doc.rust-lang.org/stable/core/ffi/struct.CStr.html#method.from_bytes_until_nul)

https://github.com/LedgerHQ/ledger-device-rust-sdk/blob/3c22c1c1b5e2d909e34409fc92cfeed775541a63/ledger_device_sdk/src/nbgl.rs#L328-L334

Also the `unwrap` may be avoided on user provided fields, and return an error instead.

**Comment by yogh333** [comment]
Ok but in that case, it means that you must work, at application level, with nul-terminated string with no nul bytes in the middle.

**Comment by dfordivam** [comment]
@yogh333 That is true even right now, unless the `unwrap()` is removed

**Comment by yogh333** [comment]
@dfordivam No, nowadays you can work with `&str` or `String`  at application level and Rust SDK will convert them seamlessly into a C compatible string with `CString::new`. Avoiding additional heap allocation as you propose is thus counter balanced with less comfort at application level, as it implies adding systematically '\0' at the end of every string to be passed to NBGL object to be displayed.
